components/renku_data_services/storage_models/core.py

```diff
@@ -15,17 +17,117 @@ class CloudStorage:
 
     storage_id: str | None = None
 
+    source_path: str
+    target_path: str
+
     @classmethod
     def from_dict(cls, data: dict) -> "CloudStorage":
         """Create the model from a plain dictionary."""
+
+        if "project_id" not in data:
+            raise errors.ValidationError(message="'project_id' not set")
+        if "configuration" not in data:
+            raise errors.ValidationError(message="'configuration' not set")
+
+        if "source_path" not in data:
+            raise errors.ValidationError(message="project_id not set")
+
+        if "target_path" not in data:
+            raise errors.ValidationError(message="project_id not set")
+
+        if "type" not in data["configuration"]:
+            raise errors.ValidationError(message="'type' not set in 'configuration'")
+
         return cls(
             project_id=data["project_id"],
             storage_id=data.get("storage_id"),
             configuration=data["configuration"],
-            storage_type=data["configuration"].get("type"),
+            storage_type=data["configuration"]["type"],
+            source_path=data["source_path"],
+            target_path=data["target_path"],
         )
 
     @classmethod
-    def from_url(cls, storage_url: str, project_id: str) -> "CloudStorage":
+    def from_url(cls, storage_url: str, project_id: str, target_path: str) -> "CloudStorage":
```

Comment:
Can we document which standard or where the documentation for this url format is? Or are we making it up ourselves?

Reply:
there's no central standard, it's based on the different formats by the different providers. I'll add some documentation.